### ISSUE TITLE
chown: /consul/config: Read-only file system

### ISSUE THREAD

**Issue by tscolari** [kelseyhightower/consul-on-kubernetes#28]
I'm getting this when I try to start the statefulset

```
NAME       READY     STATUS             RESTARTS   AGE       IP              NODE
consul-0   0/1       CrashLoopBackOff   1          12s       10.200.100.13   k8s-node-2
```

```
kubectl logs consul-0
chown: /consul/config: Read-only file system
```

I didn't see any other issue here so I'm assuming I did something really wrong,
but `/consul/config` is the mount point for the configmap, so it makes sense that it's read only I guess.

**Comment by lsgrep** [comment]
You have to change mount path from `/consul/config` to `/consul/myconfig`, then you can mount ConfigMap to this directory.  I guess mounting does not work if that dir already exists. I changed the mount path and it works.

**Comment by dinar-dalvi** [comment]
@lsgrep good one, thanks it worked for me as well. I am using k8s 1.8.10.

**Comment by mgyong** [comment]
@dinar-dalvi @lsgrep Can you guys be more explicit on the change that was done in the yaml file? I don't fully understand where precisely is share

**Comment by martwetzels** [comment]
@mgyong I just ran into the same issue. Look at the statefulset/consul.yaml file at the volumeMounts.
```
volumeMounts:
            - name: data
              mountPath: /consul/data
            - name: config
              mountPath: /consul/myconfig <-- this one was originally /consul/config
            - name: tls
              mountPath: /etc/tls
```

Don't forget to update the config-file args in the same file:
```
- "-config-file=/consul/myconfig/server.json"
```

Hope that helps.

**Comment by mgyong** [comment]
@martwetzels It works now. Thks. I forgot to update the config-file args in the same file

**Comment by innovia** [comment]
@martwetzels I'm stuck I've updated the files like you said but for some reason running again `kubectl apply -f consul.yaml` doesn;t remove the old entry it tries to add one
```
 volumeMounts:
          - mountPath: /consul/data
            name: data
          - mountPath: /consul/my-config
            name: config
          - mountPath: /consul/config
            name: config
          - mountPath: /etc/tls
            name: tls
```

i'm affraid to run kubectl delete on the file to loose my data 

can you please help?

**Comment by innovia** [comment]
so digging a little deeper shows this:
```
making atomic writer volumes (secret, configmap, downwardAPI, projected) readonly was part of the fix for #60814

Fix impact:
Secret, configMap, downwardAPI and projected volumes will be mounted as read-only volumes. Applications that attempt to write to these volumes will receive read-only filesystem errors. Previously, applications were allowed to make changes to these volumes, but those changes were reverted at an arbitrary interval by the system. Applications should be re-configured to write derived files to another location.```

anyone was able to upgrade the version of cosul?

**Comment by kelseyhightower** [comment]
Fixed on master.
